backend/main/server/app.go

```diff
@@ -709,6 +707,26 @@ func (a *App) getProposal(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	c := models.Community{ID: p.Community_id}
+	if err := c.GetCommunity(a.DB); err != nil {
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	strategy, err := models.MatchStrategyByProposal(*c.Strategies, *p.Strategy)
+	if err != nil {
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	snapshotResponse, err := a.SnapshotClient.GetSnapshotStatus(strategy.Contract)
```

Comment:
we might want to make this request conditional depending on the strategy, since only a few of the strategies require the snapshotter